tests/test_pricing_model.py

```diff
@@ -80,37 +95,32 @@ class TestCaseCalcOutGivenInFailure:
     __test__ = False  # pytest: don't test this class
 
 
-class TradeResult:
-    """A helper class that holds the result of calling a trade function"""
+@dataclass
+class TestResultCalcOutGivenInSuccess:
+    """Dataclass for calc_out_given_in test results"""
 
-    # pylint: disable=too-few-public-methods
+    without_fee_or_slippage: float
+    without_fee: float
+    element_fee: float | None
+    element_with_fee: float | None
+    hyperdrive_fee: float
+    hyperdrive_with_fee: float
 
     __test__ = False  # pytest: don't test this class
 
-    def __init__(self, without_fee_or_slippage, with_fee, without_fee, fee):
-        self.without_fee_or_slippage = without_fee_or_slippage
-        self.with_fee = with_fee
-        self.without_fee = without_fee
-        self.fee = fee
-
-    def compare_trade_results(self, other):
-        """Compares two TradeResult instances and fails if they aren't approximately equal"""
-        np.testing.assert_almost_equal(
-            self.without_fee_or_slippage, other.without_fee_or_slippage, err_msg="unexpected without_fee_or_slippage"
-        )
-        np.testing.assert_almost_equal(self.with_fee, other.with_fee, err_msg="unexpected without_fee")
-        np.testing.assert_almost_equal(self.without_fee, other.without_fee, err_msg="unexpected fee")
-        np.testing.assert_almost_equal(self.fee, other.fee, err_msg="unexpected with_fee")
-
 
-class TestHyperdrivePricingModel(unittest.TestCase):
-    """Unit tests for the Hyperdrive pricing model"""
+# NOTE: To maximally re-use the unit test cases, we pass in stretched time to
+# the Element pricing model, which is not what happens in practice. This said,
```

Comment:
What do you mean by this? We use stretched time in Element v1.

Reply:
I got this from looking at the v0.0.1 release: https://github.com/element-fi/elf-simulations/blob/v0.0.1/src/elfpy/markets.py#L229. I'll take your word for the fact that V1 uses stretched time (now that you mention it, it would have to since the time stretch is in the construction paper).